src/skmultiflow/meta/adaptive_random_forest_regressor.py

```diff
@@ -0,0 +1,497 @@
+from copy import deepcopy
+import math
+
+import numpy as np
+
+from sklearn.preprocessing import normalize
+
+from skmultiflow.core import BaseSKMObject, RegressorMixin, MetaEstimatorMixin
+from skmultiflow.drift_detection.base_drift_detector import BaseDriftDetector
+from skmultiflow.drift_detection import ADWIN
+from skmultiflow.trees.regression_hoeffding_tree import RegressionHoeffdingTree
+from skmultiflow.metrics.measure_collection import RegressionMeasurements
+from skmultiflow.utils import get_dimensions, normalize_values_in_dict, check_random_state, check_weights
+
+
+class AdaptiveRandomForestRegressor(BaseSKMObject, RegressorMixin, MetaEstimatorMixin):
+    """Adaptive Random Forest estimator.
+
+    Parameters
+    ----------
+    n_estimators: int, optional (default=10)
+        Number of trees in the ensemble.
+
+    max_features : int, float, string or None, optional (default="auto")
+        Max number of attributes for each node split.
+
+        - If int, then consider ``max_features`` features at each split.
+        - If float, then ``max_features`` is a percentage and
+          ``int(max_features * n_features)`` features are considered at
+          each split.
+        - If "auto", then ``max_features=sqrt(n_features)``.
+        - If "sqrt", then ``max_features=sqrt(n_features)`` (same as "auto").
+        - If "log2", then ``max_features=log2(n_features)``.
+        - If None, then ``max_features=n_features``.
+
+    disable_weighted_vote: bool, optional (default=False)
+        Weighted vote option.
+
+    lambda_value: int, optional (default=6)
+        The lambda value for bagging (lambda=6 corresponds to Leverage Bagging).
+
+    drift_detection_method: BaseDriftDetector or None, optional
+        (default=ADWIN(0.001))
+        Drift Detection method. Set to None to disable Drift detection.
+
+    warning_detection_method: BaseDriftDetector or None, default(ADWIN(0.01))
+        Warning Detection method. Set to None to disable warning detection.
+
+    max_byte_size: int, optional (default=33554432)
+        (`RegressionHoeffdingTree` parameter)
+        Maximum memory consumed by the tree.
+
+    memory_estimate_period: int, optional (default=2000000)
+        (`RegressionHoeffdingTree` parameter)
+        Number of instances between memory consumption checks.
+
+    grace_period: int, optional (default=50)
```

Comment:
Throughout the framework the standard value for this parameter is 200.

Reply:
For the adaptive random forest algorithm a grace period of 50 is advisable. 
There is no mention of what value was used in that ARFReg paper, still, a good option in this case is to rely on the original ARF algorithm for classification for these parameters. 

Gomes, H. M., Bifet, A., Read, J., ..., Pfahringer, B., Holmes, G., & Abdessalem, T. (2017). Adaptive random forests for evolving data stream classification. Machine Learning, 106(9-10), 1469-1495.
*"By using smaller values of GP run time (and memory usage) is increased, and also causes trees to grow deeper, which enhances the overall variability of the forest, and consequently ARF’s classification performance. For consistency, we use the same base learner configuration for all ensembles, i.e. HNBTs with **GP = 50**."*

Reply:
For consistency, we should use the same default parameters as in MOA. If there are inconsistencies with the values in the paper then a note should be added.

Reply:
It is the default in MOA: 
https://github.com/Waikato/moa/blob/67699ac060d492986b30f888a7ffa34ca2eacf3f/moa/src/main/java/moa/classifiers/meta/AdaptiveRandomForest.java#L96